Sources/Purchasing/OfferingsManager.swift

```diff
@@ -275,6 +276,58 @@ private extension OfferingsManager {
         }
     }
 
+    private func fetchProducts(withIdentifiers identifiers: Set<String>, completion: @escaping ProductsManagerType.Completion) {
+        if self.systemInfo.dangerousSettings.uiPreviewMode {
+            let mockProducts = self.createMockProducts(productIdentifiers: identifiers)
+            completion(.success(mockProducts))
+        } else {
+            self.productsManager.products(withIdentifiers: identifiers, completion: completion)
+        }
+    }
+
+    private func createMockProducts(productIdentifiers: Set<String>) -> Set<StoreProduct> {
+        let productTypes = [
+            (type: "weekly", price: 1.99, period: SubscriptionPeriod(value: 1, unit: .week)),
+            (type: "monthly", price: 5.99, period: SubscriptionPeriod(value: 1, unit: .month)),
+            (type: "yearly", price: 59.99, period: SubscriptionPeriod(value: 1, unit: .year)),
+            (type: "lifetime", price: 199.99, period: nil)
+        ]
+
+        let products = productIdentifiers.enumerated().map { index, identifier -> StoreProduct in
+            let productType = productTypes[index % productTypes.count]
```

Comment:
I wonder if we could act a bit smarter here and instead of assigning basically a random period to each product, we could try to get the [package type](https://github.com/RevenueCat/purchases-ios/blob/main/Sources/Purchasing/PackageType.swift#L23) provided by the offerings (which has the duration). In case the package type is `custom`, we could then do something "smart" like analyzing the product id for some keywords like `week`, `month`, `year`.... To assign the duration. And then if we don't find any, as a fallback, assign a random period.